src/anndata/_core/sparse_dataset.py

```diff
@@ -545,16 +546,18 @@ def append(self, sparse_matrix: ss.csr_matrix | ss.csc_matrix | SpArray) -> None
         indptr[orig_data_size:] = (
             sparse_matrix.indptr[1:].astype(np.int64) + indptr_offset
         )
-        # Clear cached property
-        if hasattr(self, "indptr"):
-            del self._indptr
 
         # indices
         indices = self.group["indices"]
         orig_data_size = indices.shape[0]
         indices.resize((orig_data_size + sparse_matrix.indices.shape[0],))
         indices[orig_data_size:] = sparse_matrix.indices
 
+        # Clear cached property
+        for attr in ["_indptr", "_indices", "_data"]:
+            if hasattr(self, attr):
+                delattr(self, attr)
```

Comment:
before it was

```py
if hasattr(self, "indptr"):  # no underscore
    del self._indptr  # underscore
```

now it checks and `del`s the same name (with underscore).

is the new test testing that this works as expected or could that change be a problem?

Reply:
Right, I think this was a bug.  So without this change, the test fails:

```python
>           assert old_array_shapes[name] != getattr(diskmtx, f"_{name}").shape
E           AssertionError: assert (101,) != (101,)
E            +  where (101,) = array([  0,   1,   1,   2,   2,   4,   4,   4,   5,   5,   5,   5,   8,\n         8,   9,  11,  12,  12,  14,  16,  18,...76,  76,  78,  78,  79,  81,  82,  83,  85,  86,\n        87,  90,  92,  94,  97,  97,  97,  97,  97, 100], dtype=int32).shape
E            +    where array([  0,   1,   1,   2,   2,   4,   4,   4,   5,   5,   5,   5,   8,\n         8,   9,  11,  12,  12,  14,  16,  18,...76,  76,  78,  78,  79,  81,  82,  83,  85,  86,\n        87,  90,  92,  94,  97,  97,  97,  97,  97, 100], dtype=int32) = getattr(CSRDataset: backend zarr, shape (200, 100), data_dtype float64, '_indptr')```

Reply:
Ancillary, but direct relevant to the PR